Proposal for new PR template (#34981)
space-wizards/space-station-14: .github/PULL_REQUEST_TEMPLATE.md

<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- Summarize what you changed here. -->

## Reason for change
<!-- Explain why it was changed. Link any relevant discussions or issues. -->

## Effect on game balance
<!-- Explain how this will affect game balance. -->
This will not affect game balance.

## Technical details

> <!-- Summarize what code you changed, if any. This will speed up your review time. -->

Reviewer: I would reformulate this part. I don't need a full list of "changed this file to do XYZ", I can look at the code for that. What should be written there are the reason for some technical decisions that have been made, so I don't have to figure it out myself, for example "we do it this way and not the other way, because..."

Reviewer 2: This should be a high level technical summary of the changes that would aid a reviewer in understanding what's going on *before* reviewing the diff.

Reviewer: Yeah, high level is the important part.

> <!-- Summarize what code you changed, if any. This will speed up your review time. -->

Reviewer 2: ```suggestion
<!-- High level technical details that would help reviewers understand the code changes you made. Explain the strategy of your changes and how they work at a high level. Required for all but the most blindingly obvious changes. -->
```

## Media

> <!-- Attach media if the PR makes changes that affect the in-game experience, such as new clothing, items, features, or important bug fixes.
> Use screenshots or recorded videos as appropriate. Small fixes and refactors are exempt. Media may be used in progress reports with credit. -->

Reviewer 2: ```suggestion
<!-- Attach screenshots or screen recordings demonstrating that new features or fixes work as intended.
Media attached here may also be used in progress reports with credit. -->
```

## Requirements
<!-- Confirm the following by placing an X in the brackets [X]: -->
- [ ] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).

> - [ ] I have added media to this PR or it does not require an ingame showcase.

Reviewer 2: ```suggestion

```

> - [ ] I have added test steps to this PR, or it does not require testing.

Reviewer 2: All PR's require testing

Author: then there should be test steps :godo:

> - [ ] I have added test steps to this PR, or it does not require testing.

Reviewer 2: ```suggestion
- [ ] I have added test steps to this PR, or sufficient testing is shown in media.
```

<!-- IMPORTANT: Not following the above will get your PR closed at the discretion of the maintainers. -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

## Test steps
<!-- Write some test steps that prove your PR works as intended. The first step is assumed to be "Load the Dev station" unless stated otherwise. This will speed up your review time. An example is included below -->
1. Use the entity spawn window to spawn Ian
2. Pet Ian

> 3. You should hear Ian bark and little love hearts appear

Reviewer 2: Better for this to come right after media.

<!-- Guidelines: https://docs.spacestation14.io/en/getting-started/pr-guideline -->

## About the PR
<!-- Summarize what you changed here. -->

## Reason for change
<!-- Explain why it was changed. Link any relevant discussions or issues. -->

> ## Effect on game balance

Author: I've broken the reason for the change and the effect on balance into two sections. This makes PRs easier to read and encourages folks to specifically explain the balance change away from "balance is bad" which the old combined section encouraged.

Author: Or rather:

"Nukie zombie ops are not fun." is a reason TO change something but it's not the impact OF changing something.

- [ ] I have added media to this PR or it does not require an ingame showcase.
- [ ] I have added test steps to this PR, or it does not require testing.
<!-- IMPORTANT: Not following the above will get your PR closed at the discretion of the maintainers. -->

## Breaking changes
<!-- List any breaking changes, including namespaces, public class/method/field changes, prototype renames; and provide instructions for fixing them.
This will be posted in #codebase-changes. -->

## Test steps
<!-- Write some test steps that prove your PR works as intended. This will speed up your review time. An example is included below -->
1. Open the game and load into the Dev station
2. Use the entity spawn window to spawn Ian
3. Pet Ian
4. You should hear Ian bark and little love hearts appear

**Change log**
<!-- Add a change log entry to make players aware of new features or changes that could affect gameplay.
Make sure to read the guidelines. Your change log must have a :cl: symbol, so our bot recognizes the changes
and adds them to the game's change log. An example change log is below.
-->
<!--

> :cl:

Author: Sorry, can you be more specific here?

> :cl:

Reviewer 3: You can do something like
"If you want to be attributed under a different name, you can do"
:cl: YourName

"For attributing multiple authors, you can do"
:cl: FirstAuthor, SecondAuthor, ThirdAuthor...

- add: Candy bowls can now be found near waiting lines.
- remove: The syndicate headset has been removed from the uplink.
- tweak: The detective’s revolver now contains cap bullets instead of lethals.
- fix: Fixed reflected projectiles dealing stamina damage unintentionally.
ADMIN:
- add: Candy bowls can now be found near waiting lines.
- remove: The syndicate headset has been removed from the uplink.
- tweak: The detective’s revolver now contains cap bullets instead of lethals.

> - fix: Fixed reflected projectiles dealing stamina damage unintentionally.

Reviewer 4: i don't know about this one, it already is in the pr guide link that is given above

Reviewer 5: People don't check it

Author: It's a nice-to-have reference.

Reviewer 3: yea people don't write good CLs

another suggestion is to note that mapping PRs should not have CLs, recently a lot of people have been putting in mapping CLs and its annoying to tell them every time to remove it or sometimes it goes through and clutters up the CL

-->

> :cl:

Author: Change log isn't automatically one word - it is in Perforce because it's the equivalent of a branch and commit - so I've used the lower-jargon equivalent (i.e. "change log" not "changelog".) But this is a nitpick and can be reverted as desired.

I've also split examples from the template.

Reviewer 6: If someone forgets to remove the empty change log prefixes from their PR and it gets merged, will they show up in the change log, or is the bot able to ignore lines which are empty?

Author: :stare:

I didn't write the blog 😥
